pennylane/workflow/execution.py

```diff
@@ -85,6 +86,17 @@
 """list[str]: allowed interface strings"""
 
 
+_CACHED_EXECUTION_WITH_FINITE_SHOTS_WARNINGS = (
+    "Cached execution with finite shots detected!\n"
+    "Note that samples as well as all noisy quantities computed via sampling "
+    "will be identical across executions. This situation arises where tapes "
+    "are executed with identical operations, measurements, and parameters.\n"
+    "To avoid this behaviour, provide 'cache=False' to the QNode or execution "
+    "function."
+)
+"""str: warning message to display when cached execution is used with finite shots"""

```

Comment:
Is this only typically expected when a user is directly using `qml.execute()`?

Reply:
Nope, @trbromley!

You can easily trigger it using a `QNode` too:
```python

import pennylane as qml

dev = qml.device("default.qubit", shots=1)

@qml.qnode(dev, cache={})
def circuit():
    return qml.expval(qml.PauliZ(0))

print(circuit())
print(circuit())
```
Running the script above gives:
```
1.0
/Users/mikhail/sc/39484/pennylane/pennylane/workflow/execution.py:462: UserWarning: Cached execution with finite shots detected!
Note that samples as well as all noisy quantities computed via sampling will be identical across executions. This situation arises where tapes are executed with identical operations, measurements, and parameters.
To avoid this behaviour, provide 'cache=False' to the QNode or execution function.
  warnings.warn(_CACHED_EXECUTION_WITH_FINITE_SHOTS_WARNINGS, UserWarning)
1.0
```